src/Build/BuildCheck/OM/ParsedItemsAnalysisData.cs

```diff
@@ -53,3 +50,45 @@ public class ParsedItemsAnalysisData : AnalysisData
 
     public ItemsHolder ItemsHolder { get; }
 }
+
+/// <summary>
+/// BuildCheck OM data representing a task executed by a project.
+/// </summary>
+public sealed class TaskInvocationAnalysisData : AnalysisData
+{
+    public record class TaskParameter(object? Value, bool IsOutput);
```

Comment:
Please add an xml comment here. Especially the `Value` might need explanation about possible types, scalar vs list values...